### ISSUE TITLE
Always asking for repo username and password

### ISSUE THREAD

**Issue by aonez** [ForkIssues/Tracker#142]
Since 1.0.58.2 update (updated today) I'm asked for the username and password of the repository on each push.

![screen shot 2017-10-26 at 11 50 26](https://user-images.githubusercontent.com/480163/32046764-39f1cd06-ba44-11e7-9c9a-33f8758b1a8f.png)

Working on OS X 10.11.6.

**Comment by aonez** [comment]
It seems that the current Fork instance can't access the Keychain item for github. After typing the username and password, it created another Keychain item. Then it just asks for the username, no further password ask.

**Comment by aonez** [comment]
Reproduced also on macOS 10.13. Gonna stick with 1.0.57.6 meanwhile this is solved.

**Comment by aonez** [comment]
After rolling back to 1.0.57.6 from 1.0.58.2, I now have the issue too on the old version.

**Comment by DanPristupov** [comment]
@aonez sorry for the inconvenience. I'm fixing the problem and going to release the hotfix soon.

**Comment by aonez** [comment]
No problem @DanPristupov, just wanted to give some feedback. Thanks!

**Comment by DanPristupov** [comment]
Fork can't recognise which username to use in this case. So you can solve the problem by defining username for your remote path.

For example, if you have `https://github.com/facebook/react.git`, you need to to add username like following: `https://username@github.com/facebook/react.git`

**Comment by aonez** [comment]
But why it was recognised before? I can see the username set on the "Account" field on the keychain item.

Just to test, there's any way to change the url from and already created repository using Fork? So I can test your workaround.

**Comment by DanPristupov** [comment]
@aonez 

TLDR: here's the updated version: http://git-fork.com/update/files/Fork-1.0.58.3.dmg. Could you please try it? It should work as you expected.

For the curious ones:
Previously (<1.0.58) Fork was using shared credentials data in KeyChain. I thought it's nice that Fork leaves in users system as less as possible. For example if user entered GitHub password in terminal in the past, Fork would ask a permission to use the same entry and not create a new one.

Sounds nice, but it turned out that sharing makes many problems.

1. Other applications can change entires.
2. Other applications can modify entry **type**, after which:
a. keychain returns 'not found' when I read it (that's why Fork asks for password)
                                                     b. keychain returns 'duplicate' when I try to save entry (that's why Fork asks for password again and again).
Yes, both 'not found' and 'duplicate' for the same entry.
3. users get very nervous when their credentials unexpectedly stop working and they seem to change nothing in Fork recently.

So, in 1.0.58 I made changes which fixed problems for some users and have broken things for another ones.

Solution. Fork should not share credential Keychain entries with other applications and use it's own ones (which are stored in the Apple Keychain storage, of course). And that's what I did in the hotfix.

**Comment by aonez** [comment]
Thanks a lot @DanPristupov, having and application password for Fork is by far the best solution. Removed the username from the url and after creating the new Keychain item (when username and password asked) works as expected.

Just gonna remove the old Keychain entry.

![untitled](https://user-images.githubusercontent.com/480163/32110076-e8eb9ea2-bb36-11e7-884a-b9ccbff2a0d7.png)

**Comment by DanPristupov** [comment]
@aonez you should not remove the old entry, because it's used by terminal git, xcode and others

**Comment by aonez** [comment]
> it's used by terminal git, xcode and others

Did not remember this 😅

**Comment by DanPristupov** [comment]
@aonez I'll keep the topic open for some time, so other users could see it.

**Comment by WingedDoom** [comment]
> TLDR: here's the updated version: http://git-fork.com/update/files/Fork-1.0.58.3.dmg. Could you please try it? It should work as you expected.

@DanPristupov thanks for a hotfixed build here! For me it kinda works, passwords get saved now, and Fork doesn't ask me to re-enter it. However it still asks for a login every time a try a remote operation. I enter the login and the password gets hooked automatically then.

**Comment by DanPristupov** [comment]
@WingedDoom could you send log to support@git-fork.com please? It will help me to investigate why this happens. The log is located at ~/Library/Caches/Fork/fork.log

**Comment by DanPristupov** [comment]
@WingedDoom late response is fine. I've received the log and investigating it. Thank you!

**Comment by jeffprism** [comment]
looks like this is back in Fork 1.51.1.0 Windows.

**Comment by DanPristupov** [comment]
@jeffprism we discuss the Windows version there: https://github.com/ForkIssues/TrackerWin/issues/862

**Comment by Sabreclaw** [comment]
its back
